m4/libmesh_metaphysicl.m4

```diff
@@ -12,6 +12,34 @@ AC_DEFUN([CONFIGURE_METAPHYSICL],
                          [AC_MSG_ERROR(bad value ${enableval} for --enable-metaphysicl)])],
                 [enablemetaphysicl=$enablenested])
 
+  AC_ARG_WITH(metaphysicl,
+              AS_HELP_STRING([--with-metaphysicl=<internal,/some/dir>],
+                             [internal: build from contrib]),
+              [AS_CASE("${withval}",
+                       [internal], [build_metaphysicl=yes],
+                       [METAPHYSICL_DIR="${withval}"
+                        build_metaphysicl=no])
+               enablemetaphysicl=yes],
+              [build_metaphysicl=yes])
+
+  AC_ARG_WITH(metaphysicl-include,
+              AS_HELP_STRING([--with-metaphysicl-include=</some/includedir>]),
+              [METAPHYSICL_INCLUDE="-I${withval}"
+               METAPHYSICL_INC="-I${withval}"
+               enablemetaphysicl=yes
+               build_metaphysicl=no],
+              [AS_IF([test "x$build_metaphysicl" = "xyes"],
+                     [METAPHYSICL_INCLUDE="-I\$(top_srcdir)/contrib/metaphysicl/src/numerics/include -I\$(top_srcdir)/contrib/metaphysicl/src/core/include -I\$(top_srcdir)/contrib/metaphysicl/src/utilities/include"
+                      METAPHYSICL_INC="-I$top_srcdir/contrib/metaphysicl/src/numerics/include -I$top_srcdir/contrib/metaphysicl/src/core/include -I$top_srcdir/contrib/metaphysicl/src/utilities/include"],
+                     [METAPHYSICL_INCLUDE="-I$METAPHYSICL_DIR/include"
+                      METAPHYSICL_INC="-I$METAPHYSICL_DIR/include"])]
+             )
+
+  dnl The above distinction is to make it easier to add
+  dnl --with-metaphysicl-lib later, but we don't care about
+  dnl metaphysicl_version(); right now everything we need from
+  dnl MetaPhysicL is header-only.
```

Comment:
Can you remind me what happens if a user configures with:
```
--with-metaphysicl
```
i.e. they forget the "=internal" part? I guess that matches the default case so that `build_metaphysicl` is set to `no` and then `METAPHYSICL_DIR` is set to "" (empty string)? I guess this would then fail somewhere in the next step since that directory is not valid...

Reply:
Yeah, and then if you have `--enable-metaphysicl-required` configure will scream and die; if not then it'll just quietly report that MetaPhysicL wasn't detected and wasn't enabled.

The with/enable distinction is standard GNU configure stuff, but it is a bit user-unfriendly, isn't it?  Someone wants to --enable-metaphysicl, but they forget the syntax and use --with-metaphysicl instead, and this has the effect of *disabling* what they want?  Yeah, let me fix that.